### DISCUSSION TITLE
What is the role of visible attribute in the object category defination?

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Knowlg/Community#36] [General Discussions]
I tried to change the value from true to false considering that it may hide the control on the form. It does not do anything. Can we hide the form controls? 

domain/action/object/category/definition/v1/read?fields=objectMetadata,forms,name

![image](https://user-images.githubusercontent.com/22250006/179499828-61d86b69-525d-403b-b5ff-909c55c5ad5f.png)

@maheshkumargangula @pallakartheekreddy can you please help me here

**Comment by pallakartheekreddy** [comment]
@pranavirabatti To hide fields from UI forms we need to use `"visible": false` only but I have cross-checked there is a bug it is still validating hidden fields as well. As a workaround, please remove `"type": "required"` object from `validations` array or empty the `"validations": []` will solve the problem.

**Reply by pranavirabatti** [reply]
`"visible": false` is not working. Am I right?

**Reply by pallakartheekreddy** [reply]
@pranavirabatti Yes, only changing `"visible": false` is not working.
If we change both `"visible": false`  & `"validations": []` then it will work and can you share a sample config which field you want to hide from the form.

**Reply by pranavirabatti** [reply]
`"editable": false` is also not working properly. I can still edit the form controls.

**Reply by pranavirabatti** [reply]
```
{
  "code": "license",
  "description": "license",
  "dataType": "text",
  "placeholder": "Select License",
  "defaultValue": "CC BY 4.0",
  "validations": [],
  "editable": false,
  "required": true,
  "visible": false,
  "label": "License",
  "renderingHints": {
    "class": "sb-g-col-lg-1 required"
  },
  "inputType": "select",
  "name": "license"
}
```

![image](https://user-images.githubusercontent.com/22250006/179515100-544c8af9-6299-480c-933c-88e6d967b521.png)

**Reply by pallakartheekreddy** [reply]
@pranavirabatti one thing, once we make get the data from category definition API we will store it in the session storage of the browser. After updating the form in category definition and if you are not able to see the latest changes Please clear the session storage of the browser and refresh the page or please try to re-login to reflect the latest changes.

**Reply by pranavirabatti** [reply]
ok. Let me try this.

**Reply by pranavirabatti** [reply]
> @pranavirabatti one thing, once we make get the data from category definition API we will store it in the session storage of the browser. After updating the form in category definition and if you are not able to see the latest changes Please clear the session storage of the browser and refresh the page or please try to re-login to reflect the latest changes.

1) Cleared the session/local storage.
2) Refresh 
3) Tried re-login

Nothing works!

**Reply by pallakartheekreddy** [reply]
@pranavirabatti Can we get into a call to check this? Please join the below link

https://meet.google.com/bzd-dbgi-iri

**Reply by pallakartheekreddy** [reply]
@pranavirabatti As discussed, We have fixed these `visible` & `editable` issues in release-4.7.0 and later and currently, you are using release-4.4.0 that's the reason it is not working for you.

**Reply by vinukumar-vs** [reply]
@pallakartheekreddy Thaks for detailed discussions. The release version of the adopter system/environment players the key role. 

@vaibhavbhuva @harishGangula @AmiableAnil Make sure we always check what version of the player/editor/microservice that adopter is using. There can be chances that the fix of the latest release will be working for us but not for the adopters because of older release.

@rhwarrier, @vrayulu

**Reply by pallakartheekreddy** [reply]
@pranavirabatti As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
